src/traverse/Indicator.ts

```diff
@@ -413,6 +418,13 @@ export class StateVariableIndicator extends FunctionDefinitionIndicator {
 
   updateFromBinding() {
     // it's possible we dont know in this fn scope whether a state is whole/owned or not, but the binding (contract scope) will
+    // add nullifyingPaths we didn't know were nullifying
+    if (this.binding.isWhole && this.isModified) {
+      this.modifyingPaths.forEach(modPath => {
+        // if not included, we add it
+        if (!this.nullifyingPaths.some(p => p.node.id === modPath.node.id)) this.addNullifyingPath(modPath);
+      })
+    }
```

Comment:
When we update each indicator with new info from the binding, sometimes we discover states are whole after we have added modifying paths. When a whole state is edited, it's nullified, so we need to add these to nullifyingPaths.